Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -1284,7 +1285,11 @@ bool RenderTheme::isWindowActive(const RenderObject& renderer) const
 
 bool RenderTheme::isChecked(const RenderObject& o) const
 {
-    return is<HTMLInputElement>(o.node()) && downcast<HTMLInputElement>(*o.node()).shouldAppearChecked();
+    if (auto* element = dynamicDowncast<HTMLInputElement>(o.node()))
+        return element->shouldAppearChecked();
+    if (auto* host = dynamicDowncast<HTMLInputElement>(o.node()->shadowHost()))
```

Comment:
What if `o.node()` returns nullptr?

The previous code was dealing with that fine. the new code would crash.